extensions/amp-user-location/0.1/test/test-config-loader.js

```diff
@@ -0,0 +1,226 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ConfigLoader} from '../config-loader';
+
+describes.fakeWin(
+  'ConfigLoader',
+  {
+    amp: {
+      extensions: ['amp-user-location'],
+    },
+  },
+  env => {
+    let win;
+    let doc;
+    let fakeAmpdoc;
+
+    beforeEach(() => {
+      win = env.win;
+      doc = win.document;
+      fakeAmpdoc = {win};
+    });
+
+    function newFakeUserLocation(opt_config, opt_src) {
+      const userLocation = doc.createElement('fake-amp-user-location');
+      doc.body.appendChild(userLocation);
+
+      if (opt_config) {
+        const configElement = win.document.createElement('script');
+        configElement.setAttribute('type', 'application/json');
+        if (typeof opt_config == 'string') {
+          configElement.textContent = opt_config;
+        } else {
+          configElement.textContent = JSON.stringify(opt_config);
+        }
+        userLocation.appendChild(configElement);
+      }
+
+      if (opt_src) {
+        userLocation.setAttribute('src', opt_src);
+      }
+
+      return userLocation;
+    }
+
+    describe('script tag config', () => {
+      it('should parse config if present', async () => {
+        const EMPTY_CONFIG = {};
+        const elementNoConfig = newFakeUserLocation();
+        const noConfigLoader = new ConfigLoader(fakeAmpdoc, elementNoConfig);
+        const emptyConfig = await noConfigLoader.getConfig();
+        expect(emptyConfig).to.deep.equal(EMPTY_CONFIG);
+
+        const elementWithConfig = await newFakeUserLocation({});
+        const configLoader = new ConfigLoader(fakeAmpdoc, elementWithConfig);
+        const config = await configLoader.getConfig();
+        expect(config).to.deep.equal(EMPTY_CONFIG);
+      });
+
+      it('should error with invalid config', async () => {
+        const invalidElement = newFakeUserLocation('this is not valid json');
+        const invalidConfigLoader = new ConfigLoader(
+          fakeAmpdoc,
+          invalidElement
+        );
+        await allowConsoleError(async () => {
+          try {
+            await invalidConfigLoader.getConfig();
+          } catch (err) {
+            expect(err.message).to.include('expected configuration');
+          }
+        });
+
+        const arrayElement = newFakeUserLocation('[]');
+        const arrayConfigLoader = new ConfigLoader(fakeAmpdoc, arrayElement);
+        await allowConsoleError(async () => {
+          try {
```

Comment:
Use a rejection expectation instead. If the code changes and this does not throw an error, you will not hit the catch below, and you will never hit the expect it contains. Something like:


```javascript
await expect(arrayConfigLoader.getConfig()).to.eventually.be.rejectedWith(/expected configuration/);
```

Reply:
Nice this is better, thanks!